beacon-chain/sync/initial-sync/blocks_fetcher_test.go

```diff
@@ -743,6 +745,200 @@ func TestBlocksFetcher_filterPeers(t *testing.T) {
 	}
 }
 
+func TestBlocksFetcher_filterScoredPeers(t *testing.T) {
+	type weightedPeer struct {
+		peer.ID
+		usedCapacity int64
+	}
+	type args struct {
+		peers           []weightedPeer
+		peersPercentage float64
+		capacityWeight  float64
+	}
+
+	batchSize := uint64(flags.Get().BlockBatchLimit)
+	tests := []struct {
+		name   string
+		args   args
+		update func(s *peers.BlockProviderScorer)
+		want   []peer.ID
+	}{
+		{
+			name: "no peers available",
+			args: args{
+				peers:           []weightedPeer{},
+				peersPercentage: 1.0,
+				capacityWeight:  0.2,
+			},
+			want: []peer.ID{},
+		},
+		{
+			name: "single peer",
+			args: args{
+				peers: []weightedPeer{
+					{"abc", 1200},
+				},
+				peersPercentage: 1.0,
+				capacityWeight:  0.2,
+			},
+			want: []peer.ID{"abc"},
+		},
+		{
+			name: "multiple peers same capacity",
+			args: args{
+				peers: []weightedPeer{
+					{"abc", 2400},
```

Comment:
i think it will be more readable if peers will have one letter name
```suggestion
					{"a", 2400},
```

Reply:
On error, our assertions use `%#v`, so output will be like `unexpected peer.ID[]{"abc", "def"}` i.e. no problems with readability. I don't have any strong opinion about this, though.

Reply:
i just thought the wanted will look more readable if it looked like b,a,c,d

Reply:
Got, it! Agree, will fix in a moment. Thanks!